amg.c

```diff
@@ -662,6 +758,10 @@ main( hypre_int argc,
    /* Finalize MPI */
    hypre_MPI_Finalize();
 
+#ifdef USE_CALIPER
+   CALI_MARK_END("main");
+   adiak_fini();
+#endif
```

Comment:
The end main region and adiak_fini calls need to move before `hypre_MPI_Finalize()`

Reply:
done